### PR TITLE
UML: 980 - Take user to home if session expired the previous calendar day, show session expired message otherwise

### PR DESCRIPTION
**Pull request by SeemaMenon**
## Purpose
https://opgtransform.atlassian.net/browse/UML-980

Fixes UML-980

## Approach

This now compares the session expiry date and current date and if current date greater than expiry date, will take user to home page.

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Refunds service_

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [X] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

**Review by Raffers — APPROVED** [review]

File: service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php [hunk 1]
```diff
@@ -31,9 +31,24 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
...
+        
         if ($session !== null && $session->get(EncryptedCookiePersistence::SESSION_EXPIRED_KEY) !== null) {
+
+            $currentDate = date("Y-m-d", $currentDatetime);
+            $expiredDate = date("Y-m-d", $sessionExpiredDatetime);
+
             $session->unset(EncryptedCookiePersistence::SESSION_EXPIRED_KEY);
-            $uri = new Uri($this->helper->generate('/session-expired'));
+            if ($currentDate > $expiredDate) {
+                $uri = new Uri($this->helper->generate('/home'));
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php:48]: Please can these urls be generated using the route name? That way the locale will work correctly

**Review by cooperaj — CHANGES_REQUESTED** [review]

File: service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php (hunk 1 above)
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php]: done

File: service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php [hunk 2]
```diff
@@ -18,22 +18,37 @@
...
     }
 
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
         /** @var SessionInterface $session */
         $session = $request->getAttribute(SessionMiddleware::SESSION_ATTRIBUTE);
+
+        $currentDatetime =  time();
+
+        if ($session !== null) {
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php:37]: Please combine this 'if' with the one below

File: service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php [hunk 3]
```diff
@@ -18,22 +18,37 @@
...
+            $sessionExpiredDatetime = $session->get(EncryptedCookiePersistence::SESSION_TIME_KEY);
+        }
+        
         if ($session !== null && $session->get(EncryptedCookiePersistence::SESSION_EXPIRED_KEY) !== null) {
+
+            $currentDate = date("Y-m-d", $currentDatetime);
+            $expiredDate = date("Y-m-d", $sessionExpiredDatetime);
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php:44]:
```suggestion
            $expiredDate = new Datetime($session->get(EncryptedCookiePersistence::SESSION_TIME_KEY));
```

File: service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php (hunk 2 above)
Comment (cooperaj) [reply service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php:37]: Something like
```suggestion
        if ($session !== null && $session->has(EncryptedCookiePersistence::SESSION_EXPIRED_KEY)
```

**Review by cooperaj — CHANGES_REQUESTED** [review]

File: service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php [hunk 4]
```diff
@@ -18,22 +18,37 @@
...
+
+        $currentDatetime =  time();
+
+        if ($session !== null) {
+            $sessionExpiredDatetime = $session->get(EncryptedCookiePersistence::SESSION_TIME_KEY);
+        }
+        
         if ($session !== null && $session->get(EncryptedCookiePersistence::SESSION_EXPIRED_KEY) !== null) {
+
+            $currentDate = date("Y-m-d", $currentDatetime);
```
[the remark this answers is not in the document]
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php:43]: OK, done

File: service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php (hunk 3 above)
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php:44]: Couldn't get this working working. Instead tried this
$sessionExpiredDatetime = $session->get(EncryptedCookiePersistence::SESSION_TIME_KEY);
$expiredDate = (new DateTime("@$sessionExpiredDatetime"))->format('Y-m-d');

Comment (cooperaj) [reply service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php:44]: Mmm. It should have worked. I wonder what the formate of the session value is. Either way I'm not sure what `"@$sessionExpiredDatetime"` will do. I'm not sure that will work.

Comment (SeemaMenon) [reply service-front/app/src/Common/src/Middleware/Session/SessionExpiredRedirectMiddleware.php]: This works on my testing. I followed this example below:
Screenshot 2020-08-26 at 09 47 03

**Review by cooperaj — APPROVED** [review]
👍 Great

**Review by Raffers — APPROVED** [review]

**Pull request merged**